server/src/main/java/org/elasticsearch/health/HealthStatus.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health;
+
+import java.util.stream.Stream;
+
+public enum HealthStatus {
+    GREEN,
+    YELLOW,
+    RED;
+
+    public static HealthStatus merge(Stream<HealthStatus> statuses) {
```

Comment:
It would be nice to add some simple unit tests for `HealthStatus.merge` just to make sure it correctly merges statuses.